src/main/java/org/tron/keystore/WalletFile.java

```diff
@@ -0,0 +1,482 @@
+package org.tron.keystore;
+
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+
+/**
+ * Ethereum wallet file.
+ */
+public class WalletFile {
+
+  private String address;
+  private Crypto crypto;
+  private String id;
+  private int version;
+
+  public WalletFile() {
+  }
+
+  public String getAddress() {
+    return address;
+  }
+
+  public void setAddress(String address) {
+    this.address = address;
+  }
+
+  public Crypto getCrypto() {
+    return crypto;
+  }
+
+  @JsonSetter("crypto")
+  public void setCrypto(Crypto crypto) {
+    this.crypto = crypto;
+  }
+
+  @JsonSetter("Crypto")  // older wallet files may have this attribute name
+  public void setCryptoV1(Crypto crypto) {
+    setCrypto(crypto);
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public void setId(String id) {
+    this.id = id;
+  }
+
+  public int getVersion() {
+    return version;
+  }
+
+  public void setVersion(int version) {
+    this.version = version;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null) {
+      return false;
+    }
+    if (o.getClass() != WalletFile.class) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Compare to "this.getClass()" instead. ([squid:S2162](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2162))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=938&issues=AWPK23iWkDisRwWBfuxG&open=AWPK23iWkDisRwWBfuxG)

[ISSUE_KEY]: AWPK23iWkDisRwWBfuxG